tests/st/bgp/test_ipip.py

```diff
@@ -287,3 +288,62 @@ def get_tunl_tx(self, host):
         match = re.search(r'RX packets:(\d+) ',
                           output)
         return int(match.group(1))
+
+    @parameterized.expand([
+        (False,),
+        (True,),
+    ])
+    def test_gce(self, with_ipip):
+        """
+        Test with and without IPIP routing on simulated GCE instances.
+        """
+        with DockerHost('host1',
+                        additional_docker_options=CLUSTER_STORE_DOCKER_OPTIONS,
+                        simulate_gce_routing=True,
+                        start_calico=False) as host1, \
+             DockerHost('host2',
+                        additional_docker_options=CLUSTER_STORE_DOCKER_OPTIONS,
+                        simulate_gce_routing=True,
+                        start_calico=False) as host2:
+
+            host1.start_calico_node()
+            host2.start_calico_node()
+
+            # We are simulating GCE instance routing, where there is a router
+            # between the instances, and each instance has a /32 address that
+            # appears not to be directly connected to any subnet.  Hence we
+            # also need to enable IP-in-IP to get from one host to the other.
+            # (In the case where we don't do this, we will expect workload
+            # connectivity to fail.)
+            if with_ipip:
```

Comment:
Sorry for the back and forth....

A big chunk of the time it takes to run the STs is performing the setup of the docker hosts.  I think it would be better to reduce this to a single test rather than running twice with the following flow:

-  create workloads
-  check connectivity fails
-  turn on IPIP
-  check connectivity passes

that way we only have to set up the hosts once for this test *and* we get the added bonus of testing that you can enable IPIP after creating the workloads and it should kick into life.